crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -37,6 +39,57 @@ impl DynamicScene {
         Self::from_world(&scene.world, type_registry)
     }
 
+    /// Create a Dynamic Scene with specific entities.
+    ///
+    /// The generic parameter is used as a `Query` filter.
+    ///
+    /// The created scene will include only the entities that match the query
+    /// filter provided. All components that impl `Reflect` will be included.
+    pub fn from_query_filter<F>(world: &mut World) -> Self
+    where
+        F: ReadOnlyWorldQuery + 'static,
+    {
+        let mut ss = SystemState::<Query<Entity, F>>::new(world);
+
+        let type_registry = world
+            .get_resource::<AppTypeRegistry>()
+            .expect("The World provided for scene generation does not contain a TypeRegistry")
+            .read();
+
+        let q = ss.get(world);
+
+        let entities = q
+            .iter()
+            .map(|entity| {
+                let get_reflect_by_id = |id| {
+                    world
+                        .components()
+                        .get_info(id)
+                        .and_then(|info| type_registry.get(info.type_id().unwrap()))
```

Comment:
Can we handle this error more robustly than via unwrapping?

Reply:
IDK, possibly i'll see what I can do!